src/modules/model.js

```diff
@@ -36,90 +40,94 @@ module.exports = ({
   };
   const get = async ({ id, fields }) => {
     precheck();
-    const splitFields = objectFields.split(fields);
-    const resp = await aws.call('dynamodb:getItem', {
-      TableName: tableName,
-      ExpressionAttributeNames: generateExpressionAttributeNames(splitFields),
-      ProjectionExpression: splitFields.map(f => `#${f.toUpperCase()}`).join(','),
-      Key: { id: { S: id } },
-      ConsistentRead: true
-    });
-    if (resp.Item === undefined) {
-      throw EntryNotFound({ id });
+    let resp;
+    try {
+      resp = await mapper.get(Object.assign(new Model(), { id }), {
+        projection: objectFields.split(fields),
+        readConsistency: 'strong'
+      });
+    } catch (err) {
+      if (err.name === 'ItemNotFoundException') {
+        throw ItemNotFound({ id });
+      }
+      throw err;
     }
     await internalCallback({ id, actionType: 'get' });
-    return dynamodbConverter.unmarshall(resp.Item);
+    return resp;
   };
 
   return {
     get,
     create: async ({ id, data, fields }) => {
       precheck();
-      const resp = await aws.call('dynamodb:putItem', {
-        ConditionExpression: 'id <> :id',
-        ExpressionAttributeValues: { ':id': { S: id } },
-        TableName: tableName,
-        Item: dynamodbConverter.marshall(Object.assign({}, data, { id }))
-      }, {
-        expectedErrorCodes: ['ConditionalCheckFailedException']
-      });
-      if (resp === 'ConditionalCheckFailedException') {
-        throw EntryExists({ id });
+      try {
+        await mapper.put(Object.assign(new Model(), data, { id }), {
```

Comment:
We might want to wrap `Model` in a closure and have dedicated functions to update etc. I think that will make it a lot cleaner and future proof